components/Discussion/hooks/actions/useSubmitCommentHandler.ts

```diff
@@ -0,0 +1,91 @@
+import { FetchResult } from '@apollo/client'
+import uuid from 'uuid/v4'
+import { toRejectedString } from '../../graphql/utils'
+import Optional from '../../../../lib/types/Optional'
+import {
+  SubmitCommentMutationResult,
+  useSubmitCommentMutation
+} from '../../graphql/mutations/SubmitCommentMutation.graphql'
+import { useDiscussion } from '../../context/DiscussionContext'
+import {
+  DISCUSSION_QUERY,
+  DiscussionQuery,
+  DiscussionQueryVariables
+} from '../../graphql/queries/DiscussionQuery.graphql'
+import produce from 'immer'
+import { mergeComment } from '../../graphql/store'
+
+export type SubmitCommentHandlerFunction = (
+  content: string,
+  tags: string[],
+  options: {
+    discussionId: string
+    parentId: Optional<string>
+  }
+) => Promise<FetchResult<SubmitCommentMutationResult>>
+
+function useSubmitCommentHandler(): SubmitCommentHandlerFunction {
+  const [submitCommentMutation] = useSubmitCommentMutation()
+
+  const { orderBy, focusId, activeTag } = useDiscussion()
+
+  return async (
+    content: string,
+    tags: string[],
+    {
+      discussionId,
+      parentId
+    }: {
+      discussionId: string
+      parentId: Optional<string>
+    }
+  ) => {
+    const id = uuid()
+
+    return submitCommentMutation({
+      variables: {
+        id,
+        content,
+        discussionId,
+        parentId,
+        tags
+      },
+      // Write the result of the query into the DiscussionQuery cache
+      update: (cache, { data: { submitComment: comment } }) => {
+        const variables: DiscussionQueryVariables = {
+          discussionId,
+          orderBy: orderBy,
+          depth: 3,
```

Comment:
this should probably not be hard coded and passed from where it is either defined as 1 or 3.